.github/workflows/build-ultralite.yml

```diff
@@ -0,0 +1,111 @@
+# Build directly on the GitHub runner with caching
+name: build-ultralite
+on:
+  workflow_dispatch:
+  workflow_call:
+
+concurrency:
+  group: build-ultralite-${{github.ref}}-${{github.event.pull_request.number || github.run_number}}-${{github.workflow}}
+  cancel-in-progress: true
+
+jobs:
+  linux:
+    continue-on-error: true
+    name: ultralite-ubuntu
+    env:
+      CCACHE_DIR: "${{github.workspace}}/.ccache"
+      CCACHE_MAXSIZE: "20Mi"
+      CMAKE_PRESET: ultralite
+    runs-on: ubuntu-latest
+    steps:
+      - name: Install dependencies
+        run: |
+          sudo apt-get -q -y update
+          sudo apt-get -q -y install \
+            ccache cmake ninja-build
+      - name: Check out Celeritas
```

Comment:
It feels like from there on, the code is not specific to `ultralite` (except for the cache name), if this is correct then in this PR and/or a subsequent PR we should consider factoring the code.

Reply:
It's effectively the same as the `fast` build yes, but I'm not sure how refactoring could improve much...

Reply:
Fair enough.